tket/src/Transformations/BasicOptimisation.cpp

```diff
@@ -238,8 +238,12 @@ static bool commute_singles_to_front(Circuit &circ) {
             if (type == EdgeType::Quantum) {
               boundary_edge = circ.get_last_edge(current_v, current_e);
             } else {
-              // There should not be any Classical edges for conditional ops
-              TKET_ASSERT(type == EdgeType::Boolean);
+              if (type != EdgeType::Boolean) {
+                // There should not be any Classical edges for conditional ops
+                TKET_ASSERT(type == EdgeType::Classical);
+                throw CircuitInvalidity(
+                    "Malformed conditional operation: found Classical wire");
```

Comment:
Sorry to quibble again, but is it actually malformed? I may be missing something but I still don't see why this couldn't arise in some legitimate way?

Reply:
Turns out this was very easy.

Reply:
Thanks! I had missed the fact that `ends_commute()` guarantees this condition.